src/fetchWebSetup.ts

```diff
@@ -42,8 +64,29 @@ export function getCachedWebSetup(options: any): WebConfig | undefined {
  */
 export async function fetchWebSetup(options: any): Promise<WebConfig> {
   const projectId = getProjectId(options, false);
-  const res = await apiClient.get<WebConfig>(`/projects/${projectId}/webApps/-/config`);
+
+  // Try to determine the appId from the default Hosting site, if it is linked.
+  let hostingAppId: string | undefined = undefined;
+  try {
+    const sites = await hostingApiClient.get<ListSitesResponse>(`/projects/${projectId}/sites`);
+    const defaultSite = sites.body.sites.find((s) => s.type === "DEFAULT_SITE");
```

Comment:
Does the emulator support non-default sites?

Reply:
At the moment, no.  One day!